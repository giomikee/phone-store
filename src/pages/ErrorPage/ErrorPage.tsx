import { Box, Button, Typography } from '@mui/material';
import BugReportIcon from '@mui/icons-material/BugReport';
import UndoIcon from '@mui/icons-material/Undo';
import { Link } from 'react-router-dom';
import { PAGES } from '../../constants';

const ErrorPage = () => {
    return (
        <>
            <BugReportIcon 
                color='error'
                sx={{
                    height: '150px',
                    width: '100%'
                }}
            />

            <Box sx={{ flexGrow: 1, textAlign: 'center'}}>
                <Typography 
                    component='h2' 
                    gutterBottom
                    variant='h5'
                >
                    Error técnico detectado. Intentélo de nuevo más tarde.
                </Typography>

                <Link to={PAGES.home}>
                    <Button
                        startIcon={<UndoIcon />}
                        variant='outlined'
                    >
                        Volver al inicio
                    </Button>
                </Link>
            </Box>
        </>
    )
}

export default ErrorPage