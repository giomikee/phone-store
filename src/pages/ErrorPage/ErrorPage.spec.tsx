import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { BrowserRouter } from 'react-router-dom';
import ErrorPage from './ErrorPage';

describe('Given ErrorPage component', () => {
    describe('When the component is rendered', () => {
        it('Then it should match the snapshot', () => {
            const { container } = render(<ErrorPage />, { wrapper: BrowserRouter });
            expect(container).toMatchSnapshot();
        });
    });
});