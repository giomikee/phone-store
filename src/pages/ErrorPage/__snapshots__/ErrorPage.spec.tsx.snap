// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given ErrorPage component > When the component is rendered > Then it should match the snapshot 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-q2ex3j-MuiSvgIcon-root"
    data-testid="BugReportIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5s-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20zm-6 8h-4v-2h4zm0-4h-4v-2h4z"
    />
  </svg>
  <div
    class="MuiBox-root css-17ls7xt"
  >
    <h2
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1f9xtph-MuiTypography-root"
    >
      Error técnico detectado. Intentélo de nuevo más tarde.
    </h2>
    <a
      href="/"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1blvszr-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="UndoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8"
            />
          </svg>
        </span>
        Volver al inicio
      </button>
    </a>
  </div>
</div>
`;
